File: Python/String-2/count_code.py
"""
Return the number of times that the string "code" appears anywhere in the given string, except we'll accept any letter for the 'd', so "cope" and "cooe" count.

count_code('aaacodebbb') → 1
count_code('codexxcode') → 2
count_code('cozexxcope') → 2

@author unobatbayar
"""

def count_code(str):
  counter = 0
  #we can ingore third substring 'd' because it can by any letter
  for i in range(len(str)-3):
    if str[i:i+2] == "co" and str[i+3] == "e":
        counter = counter + 1
  return counter
  
  #we can put 'code', 'cope' and 'cooe' etc in array so we can search in double loop
  #but is not efficient, it's just another way of doing it

File: CodingBat/Java/String-2/countCode.java
/*
Return the number of times that the string "code" appears anywhere in the given string, except we'll accept any letter for the 'd', so "cope" and "cooe" count.

countCode("aaacodebbb") → 1
countCode("codexxcode") → 2
countCode("cozexxcope") → 2
@author unobatbayar 
*/

public int countCode(String str) {
    char[] alphabet = {'a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z'};
    int length = str.length();
    
    for(int i = 0; i<26; i++){
      String temp = "co" + alphabet[i] + "e";
      str = str.replace(temp, "");  
    }
    
    length = length - str.length();    
    return length/4;
  }
  